### ISSUE TITLE
Consider re-enabling some skipped tests

### ISSUE THREAD

**Issue by pranavkm** [aspnet/Caching#324]
The repo has 42 skipped tests that results in 84 build warnings being produced as part of building. Both of these are because the test expects infrastructure (Redis Server \ Sql Server) to be present on the machine.

Could we instead 

a) convert these in to conditional facts that attempt to check if the infrastructure exists (for instance all our build agents have Sql Server installed)? 
b) Just skip these from test discovery via `ExcludeFromTests` 
c) Delete them because what's the point of having tests that don't run regularly?

**Comment by natemcmaster** [comment]
+1. The SQL Server tests haven't run in a long time. cc @muratg can we triage this one?

**Comment by muratg** [comment]
I agree that we need to do something. We can possibly move the tests that require additional resources on the machine (redis/sql/etc) to a separate suite.  We'd possibly run some of those tests with the Antares test runs that we're planning to do daily.
